aiida_vasp/calcs/workchains/restart.py

```diff
@@ -126,41 +96,45 @@ def __init__(self, *args, **kwargs):
     def define(cls, spec):
         super(BaseRestartWorkChain, cls).define(spec)
         spec.input(
-            'max_iterations',
+            'restart.max_iterations',
             valid_type=Int,
             default=Int(5),
+            required=False,
             help="""
-            the maximum number of iterations the workchain will attempt to get the calculation to finish successfully
+            the maximum number of iterations the BaseRestartWorkChain will attempt to get the calculation to finish successfully
             """)
         spec.input(
-            'clean_workdir',
+            'restart.clean_workdir',
             valid_type=Bool,
             default=Bool(False),
+            required=False,
             help="""
-            when set to True, the work directories of all called calculation will be cleaned at the end of workchain execution
+            when set to True, the work directories of all called calculation will be cleaned at the end of BaseRestartWorkChain execution
             """)
 
-    def setup(self):
+    def init_context(self):
         """Initialize context variables that are used during the logical flow of the BaseRestartWorkChain."""
         self.ctx.unexpected_failure = False
         self.ctx.submission_failure = False
         self.ctx.restart_calc = None
         self.ctx.is_finished = False
         self.ctx.iteration = 0
+        self.ctx.max_iterations = self.inputs.restart.max_iterations.value
 
         return
 
-    def should_run_calculation(self):
+    def run_calculations(self):
```

Comment:
Some might find this name confusing, expecting it to cause calculations to be run

Reply:
Same with run...some think that will execute some calculations, hence name change. I think `should_run_calculations` is less confusing than `run_calculations`, otherwise I would have left it. But it is no biggie. If you insist let us use `run_calculations`. Guess this is anyway going to be moved to Aiida at some stage and will probably receive its own polish.